src/catchup/CatchupWork.cpp

```diff
@@ -242,6 +242,13 @@ CatchupWork::runCatchupStep()
     }
 
     auto const& has = mGetHistoryArchiveStateWork->getHistoryArchiveState();
+    if (has.networkPassphrase != mApp.getConfig().NETWORK_PASSPHRASE)
```

Comment:
Won't this trip when catching up to a HAS that has an empty networkPassphrase (from older versions)?